back/src/main/java/com/cocktailpick/back/tag/service/TagService.java

```diff
@@ -35,8 +38,19 @@ public void saveAll(MultipartFile file) {
 	}
 
 	@Transactional(readOnly = true)
-	public List<TagResponse> findAllTags() {
-		return TagResponse.listOf(tagRepository.findAll());
+	public List<TagResponse> findTags(TagType tagType, Integer size) {
+		List<Tag> tags = Optional.ofNullable(tagType)
+			.map(tagRepository::findByTagType)
+			.orElseGet(tagRepository::findAll);
+		return TagResponse.listOf(findTagsBySize(tags, size));
+	}
+
+	private List<Tag> findTagsBySize(List<Tag> tags, Integer size) {
+		if (Objects.isNull(size) || size >= tags.size()) {
+			return tags;
+		}
```

Comment:
if문의 조건문에 or 연산자가 있어서 읽기 어려워요. 

```suggestion
		if (Objects.isNull(size)) {
			return tags;
		}
		if (size >= tags.size()) {
		        return tags
		}
```
로 나눠주는 것이 더 좋을 것 같습니다.

또, size >= tags.size() 를 왜 해주어야하는지 의도 파악하는 것도 힘이 들었어요. 메서드로 추출하고 메서드 이름으로 의도를 나타내는 것이 좋아보입니다.

Reply:
같은 tags를 반환하는데 조건문을 나누어주니 중복이 생기는 것 같아 같이 두었습니다. 또 Objects.isNull(size)를 통해서 뒤 조건문의 null 방어도 같이 해주고자 하는 의도도 있었습니다. 다만 의도 파악이 힘든 부분은 메서드 추출을 통해 해결하겠습니다.